src/dune_pkg/opam.mli

```diff
@@ -48,10 +48,30 @@ module Summary : sig
   type t
 
   (** A message listing selected packages *)
-  val selected_packages_message : t -> User_message.t
+  val selected_packages_message :
+    t -> lock_dir_path:Path.Source.t -> User_message.t
+end
+
+module Prefer_versions : sig
```

Comment:
I would prefer the name `Version_preference`. Although this is a personal preference that you may ignore

Reply:
or give an extra enterprise-ready alternative, `Version_policy`